src/components/calcite-picker-item/calcite-picker-item.tsx

```diff
@@ -0,0 +1,179 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Method,
+  Prop,
+  State,
+  Watch,
+  h
+} from "@stencil/core";
+import {
+  checkSquare16,
+  circle16F,
+  circleFilled16F,
+  handleVertical24,
+  square16
+} from "@esri/calcite-ui-icons";
+import { CSS } from "./resources";
+import { ICON_TYPES } from "../calcite-picker/resources";
+import CalciteIcon from "../utils/CalciteIcon";
+
+@Component({
+  tag: "calcite-picker-item",
+  styleUrl: "./calcite-picker-item.scss",
+  shadow: true
+})
+export class CalcitePickerItem {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Prop({ reflect: true }) editing = false;
+
+  @Prop() metadata: object;
+
+  @Prop() selected = false;
+
+  @Watch("selected")
+  selectedWatchHandler(newValue, oldValue) {
+    if (this.isSelected !== newValue) {
+      this.isSelected = newValue;
+      this.emitToggleEvent();
+    }
+  }
+
+  @Prop({ reflect: true }) icon: ICON_TYPES | null = null;
+
+  @Prop({ reflect: true }) textHeading: string;
+
+  @Prop({ reflect: true }) textDescription: string;
+
+  @Prop({ reflect: true }) value: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  @State() isSelected = this.selected;
+
+  dir: "rtl" | "ltr";
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    this.dir = this.el.closest('[dir="rtl"]') ? "rtl" : "ltr";
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerItemToggle: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async toggle(coerce?: boolean, emit = false) {
+    this.isSelected = typeof coerce === "boolean" ? coerce : !this.isSelected;
+    if (emit) {
+      this.emitToggleEvent();
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  iconClickHandler(event: MouseEvent) {
+    this.isSelected = !this.isSelected;
+    this.emitToggleEvent(event.shiftKey);
```

Comment:
It's still a little odd that the `shiftKey` is captured here since it seems like more of a concern of the `calcite-picker` whether or not the shiftKey matters.

Reply:
doesn't hurt anything.